docs/source/guides/reference/chapters/avocado_instrumented.rst

```diff
@@ -0,0 +1,101 @@
+    executed to make clean up after testing.
+
+FINISHED:
+    This state is the end of the test's lifecycle, and the test
+    is in this state when the testing and clean up is finished.
+    during this phase, the test results are reported to runner and
+    the testing is finished.
+
+The whole Test lifecycle is represented by diagram::
+
+                    +--------+
+                    |        |
+                    |  INIT  +------------+
+                    |        |            |
+                    +----+---+            |
+                         |                |
+                         |                |
+                     Test|initialized     |
+                         |                |
+                         |            Test|skipped
+                    +----v----+           |
+                    |         |           |
+                    |  SETUP  +-----------+---+
+                    |         |           |   |
+                    +----+----+           |   |
+                         |                |   |
+                         |                |   |
+                     Test|setuped         |   |
+                         |                |   |
+                         |                |   |
+                    +----v---+            |   |
+                    |        |            |   |
+        +-----------+  TEST  <------------+   |
+        |           |        |                |
+        |           +----+---+           Setup|timeout
+        |                |                    |
+        |     Test finished or timeout        |
+        |                |                    |
+        |                |                    |
+        |         +------v-----+              |
+        |         |            |              |
+    Test|skipped   |  TEARDOWN  <--------------+
+        |         |            |
+        |         +------+-----+
+        |                |
+        |                |
+        | Teardown finished or timeout
+        |                |
+        |                |
+        |         +------v-----+
+        |         |            |
+        +--------->  FINISHED  |
+                  |            |
+                  +------------+
+
+Timeouts
+~~~~~~~~
+
+The avocado instrumented tests can be affected by three types of
+timeouts. Test timeout, Task timeout and Job timeout. All of these
+timeouts behaves the same from the point of view of Test’s lifecycle,
+but the lifecycle will change base on in which state the timeout was
+reached. Here is a description of different Test’s lifecycles base
+on the interruptions:
+
+Timeout in SETUP
+    INIT -> SETUP(interrupted) -> TEARDOWN -> FINISHED (ERROR)
+
+Timeout in TEST
+    INIT -> SETUP -> TEST(interrupted) -> TEARDOWN -> FINISHED (INTERRUPTED)
+
+Timeout in TEARDOWN
+    INIT -> SETUP -> TEST -> TEARDOWN(interrupted) -> FINISHED (ERROR)
+
+.. note:: If Task or Job timeout are reached after some other
```

Comment:
I believe this will read better:

"If either the Task or Job timeout is reached...".